main.py

```diff
@@ -44,23 +45,86 @@ def transform_value(sensor_name, name, value):
     return transform_names[name], transform_funcs[sensor_name, name](value)
 
 
+def start_publishing(args, plugin):
+    """
+    start_publishing begins sampling and publishing iio and transformed env data
+    """
+    sch = sched.scheduler(time.time, time.sleep)
+
+    def sample_and_publish_task(scope, delay):
+        sch.enter(delay, 0, sample_and_publish_task, kwargs={
```

Comment:
so i am confused on how you have a `sch.enter` inside of `sample_and_publish_task`... are u sure?

Reply:
We discussed this offline, but I'll post a summary for future reference:

The enter function is "single shot" and does not provide support for repeated tasks. I call it from within the function to schedule it again in the future.